### PR TITLE
AIRO-1648 Fix step size in ToImageMsg() Function

### LINKED ISSUE #246: Incorrect step size for Image message in ToImageMsg()
**Issue and Steps to Reproduce** 
Adding a camera publisher to  unity project from https://github.com/Unity-Technologies/Robotics-Nav2-SLAM-Example using 
`ImageMsg rosImage = mainCameraTexture.ToImageMsg(header);`
causes error while using rqt or image_view node.
`ImageView.callback_image() while trying to convert image from 'rgba8' to 'rgb8' an exception was thrown (Image is wrongly formed: step < width * byte_depth * num_channels  or  4 != 691 * 1 * 4)
`

I think it is due to incorrect size here in this line of code, which just sends step=4 irrespective of width of image.
https://github.com/Unity-Technologies/ROS-TCP-Connector/blob/7f055108f43ab11ff297175a52af5d30fbac9d57/com.unity.robotics.ros-tcp-connector/Runtime/MessageGeneration/MessageExtensions.cs#L747

Problem can be solved by manually assigning step like 
`image_step = 4;`
`img_msg.step = image_step * (uint) frame_width;` 

I will try to reproduce and fix error for different texture types, And open a PR with correct code of `ToImageMsg()`. Although I am not sure if this is specific to ros2 or a general bug.

### PR DESCRIPTION
**Pull request by Macbull**
## Proposed change(s)

Fixes incorrect step size for image message in ToImageMsg() function, which was causing error while trying to view Image topic in rqt, image_view and rviz.

### Useful links (GitHub issues, JIRA tickets, forum threads, etc.)

Closes #246 

### Types of change(s)

- [x] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other (please describe)

## Testing and Verification

Manually checked that image shows up in rqt for each format of texture. 

### Test Configuration:
- Unity Version: [Unity 2020.3.8f1]
- Unity machine OS + version: [Windows 10]
- ROS machine OS + version: [ubuntu desktop-lxde-vnc:focal, ROS Galactic]
- ROS–Unity communication: [Docker]

## Checklist
- [x] Ensured this PR is up-to-date with the `dev` branch
- [x] Created this PR to target the `dev` branch
- [x] Followed the style guidelines as described in the [Contribution Guidelines](https://github.com/Unity-Technologies/ROS-TCP-Connector/blob/main/CONTRIBUTING.md)
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [Changelog](https://github.com/Unity-Technologies/ROS-TCP-Connector/blob/dev/com.unity.robotics.ros-tcp-connector/CHANGELOG.md) and described changes in the [Unreleased section](https://github.com/Unity-Technologies/ROS-TCP-Connector/blob/dev/com.unity.robotics.ros-tcp-connector/CHANGELOG.md#unreleased)
- [ ] Updated the documentation as appropriate

## Other comments

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/ROS-TCP-Connector?pullRequest=247) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/ROS-TCP-Connector?pullRequest=247) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/ROS-TCP-Connector?pullRequest=247) it.</sub>

File: com.unity.robotics.ros-tcp-connector/Runtime/Extensions/MessageExtensions.cs [hunk 1]
```diff
@@ -700,30 +700,35 @@ public static ImageMsg ToImageMsg(this Texture2D tex, HeaderMsg header)
         {
             byte[] data = null;
             string encoding = "rgba8";
+            int step = 4 * tex.width;
```
Comment (mrpropellers) [inline com.unity.robotics.ros-tcp-connector/Runtime/Extensions/MessageExtensions.cs:703]: Can these values be left uninitialized here?  It seems like their assignment depends on the texture format, and the default assignments are handled in the switch statement, so whatever is assigned here gets discarded anyway, right?

**Review by mrpropellers — CHANGES_REQUESTED** [review]
Hey @Macbull thanks for submitting this fix!  Requesting a small change to help make this snippet more "self-documenting."

File: com.unity.robotics.ros-tcp-connector/Runtime/Extensions/MessageExtensions.cs (hunk 1 above)
Comment (Macbull) [reply com.unity.robotics.ros-tcp-connector/Runtime/Extensions/MessageExtensions.cs]: True. I initialized it to follow existing pattern of line 702. I have updated code to leave both encoding and step uninitialized.

**Comment by mrpropellers** [comment]
Thanks for the changes @Macbull! I'll shepherd this the rest of the way through.

**Pull request merged**